toolkit/tools/pkg/imagecustomizerlib/customizepackages.go

```diff
@@ -37,167 +29,94 @@ var (
 	ErrPackageInstall             = NewImageCustomizerError("Packages:Install", "failed to install packages")
 	ErrPackageCacheClean          = NewImageCustomizerError("Packages:CacheClean", "failed to clean tdnf cache")
 	ErrMountRpmSources            = NewImageCustomizerError("Packages:MountRpmSources", "failed to mount RPM sources")
-
-	tdnfOpLines = []string{
-		"Installing/Updating: ",
-		"Removing: ",
-	}
-
-	tdnfSummaryLines = []string{
-		"Installing:",
-		"Upgrading:",
-		"Removing:",
-	}
-
-	tdnfTransactionError = regexp.MustCompile(`^Found \d+ problems$`)
-
-	// Download log message.
-	// For example:
-	//   jq 6% 15709
-	tdnfDownloadRegex = regexp.MustCompile(`^\s*([a-zA-Z0-9\-._+]+)\s+\d+\%\s+\d+$`)
 )
 
-func addRemoveAndUpdatePackages(ctx context.Context, buildDir string, baseConfigPath string, config *imagecustomizerapi.OS,
-	imageChroot *safechroot.Chroot, toolsChroot *safechroot.Chroot,
-	rpmsSources []string, useBaseImageRpmRepos bool, snapshotTime string,
-) error {
-	var err error
-
-	ctx, span := otel.GetTracerProvider().Tracer(OtelTracerName).Start(ctx, "configure_packages")
-	defer span.End()
-
-	if snapshotTime == "" {
-		snapshotTime = string(config.Packages.SnapshotTime)
-	}
-
-	tdnfChroot := imageChroot
+// executePackageManagerCommand runs a package manager command with proper chroot handling
+func executePackageManagerCommand(args []string, imageChroot *safechroot.Chroot, toolsChroot *safechroot.Chroot, distroConfig distroHandler) error {
```

Comment:
`executePackageManagerCommand` is specific to tdnf/dnf. So, it shouldn't live in the `customizepackages.go` file. Also, you should give it a more specific name. For example, `executeDnf`.

Similar comment for other functions in this file.